assets/js/modules/adsense/util/url.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * URL utilities.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { parse as pslParse } from 'psl';
+
+/**
+ * WordPress dependencies
+ */
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Returns the URL for creating a new AdSense account.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object}             options         Options for generating the URL.
+ * @param {(string|undefined)} options.siteURL The initial site URL to create the account for.
+ * @return {string} AdSense URL to create a new account.
+ */
+export const getCreateAccountURL = ( { siteURL } = {} ) => {
+	const baseURL = 'https://www.google.com/adsense/signup/new';
+	const queryParams = {
+		// TODO: Check which of these parameters are actually required.
+		source: 'site-kit',
+		utm_source: 'site-kit',
+		utm_medium: 'wordpress_signup',
+	};
+	if ( siteURL ) {
+		queryParams.url = siteURL;
+	}
+	return addQueryArgs( baseURL, queryParams );
+};
+
+/**
+ * Returns the URL to an AdSense account's overview page.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object}             options           Options for generating the URL.
+ * @param {(string|undefined)} options.accountID The AdSense account ID.
+ * @param {(string|undefined)} options.userEmail The Google account email address. Relevant
+ *                                    for users with multiple Google accounts.
```

Comment:
This indent looks off, but it's minor.

Reply:
BTW all these indent issues are from when I changed all our `?string` to `(string|undefined)` and forgot to update the indentation. 😆

Reply:
We just can't win when it comes to our docblocks, eh? 😆